src/crimson/osd/osd.cc

```diff
@@ -57,8 +58,9 @@ OSD::OSD(int id, uint32_t nonce,
     mgrc{new ceph::mgr::Client{public_msgr, *this}},
     heartbeat{new Heartbeat{*this, *monc, hb_front_msgr, hb_back_msgr}},
     heartbeat_timer{[this] { update_heartbeat_peers(); }},
-    store{std::make_unique<ceph::os::CyanStore>(
-      local_conf().get_val<std::string>("osd_data"))}
+    store{std::unique_ptr<ceph::os::FuturizedObjectStore> (ceph::os::FuturizedObjectStore::create(
```

Comment:
Yeah, returning `std::unique_ptr` from the factory is something really worth consideration.